### PR TITLE
Added `RESTResource` Python class.

### PR DESCRIPTION
**Pull request by sebastian-philipp**
This is the base class for providing a RESTful interface to a resource.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/restresource.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import json
+
+import cherrypy
+
+
+def _takes_json(func):
+    def inner(*args, **kwargs):
+        body = cherrypy.request.body.read()
+        print('body={}'.format(body))
```
Comment (rjfd) [inline src/pybind/mgr/dashboard_v2/restresource.py]: swap this by a debug log message?

**Review by rjfd — APPROVED** [review]
lgtm

File: src/pybind/mgr/dashboard_v2/restresource.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/restresource.py]: Removed that line.

**Review by p-se — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/restresource.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    curl http://127.0.0.1:8080/foo
+    curl http://127.0.0.1:8080/foo/0
+
+    """
+
+    _cp_config = {
+                  'request.error_page': {'default': json_error_page},
+                  }
+
+    def _no_implemented(self, is_element):
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard_v2/restresource.py]: I think `_not_implemented` is more intuitive as a name here.

**Review by LenzGr — APPROVED** [review]
lgtm otherwise, thanks!

**Review by votdev — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/restresource.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/restresource.py]: fixed

**Pull request merged**